zebra-state/src/in_memory.rs

```diff
@@ -0,0 +1,119 @@
+use super::{Request, Response};
+use futures::prelude::*;
+use std::{
+    collections::{BTreeMap, HashMap},
+    error::Error,
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tower::{buffer::Buffer, Service};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    types::BlockHeight,
+};
+
+#[derive(Default)]
+struct ZebraState {
```

Comment:
I wonder if this might be better to split out / rename this into a substructure like `BlockIndex` (naming tbd) and have the `ZebraState` hold onto a `BlockIndex`. The idea would be that since this code has to maintain some invariants about the two hashmaps, the maintenance of those invariants should be as tightly scoped as possible, and we will probably want to add additional state later, whose code probably shouldn't have to consider those invariants. WDYT?

Reply:
@hdevalence we will need to do queries about particular transactions too, right? We may need to namespace some of `ZebraState` methods etc